doc/book/cookbook.md

```diff
@@ -228,7 +228,55 @@ return [
                     'ValidationMiddleware',
                 ],
             ],
-        ], 
+        ],
     ],
 ];
 ```
+
+## How can I set custom 404 page handling?
+
+We can set custom 404 page handling instead of [using the final handler](http://zend-expressive.readthedocs.org/en/latest/error-handling/) for logging or other purpose with create new middleware that we want to intercept if no other middleware has executed, and is indicate a "not found" situation.
```

Comment:
Two things:

- First, use a relative link here: `../error-handling/`; I  can fix that on merge.
- Second, the example you have below is not actually by-passing the final handler. What you're doing below is setting the response status, and, by calling `$next()` with an error condition, delegating to the next error middleware present and/or the final handler itself. As such, the verbiage above is misleading, and/or the example below is misleading.

I think we need to clarify which approach you want to illustrate, and make changes to ensure the narrative and example are consistent.